db/src/db.rs

```diff
@@ -995,6 +1009,57 @@ pub fn update_partition_map(conn: &rusqlite::Connection, partition_map: &Partiti
     Ok(())
 }
 
+fn metadata_assertions(conn: &rusqlite::Connection, tx_id: Entid, comitted: bool) -> Result<Vec<(Entid, Entid, TypedValue, bool)>> {
+    let (params, sql_stmt) = match comitted {
+        true => {
+            (
+                vec![&tx_id as &ToSql],
+                format!(r#"
+                    SELECT e, a, v, value_type_tag, added
+                    FROM transactions
+                    WHERE tx = ? AND a IN {}
```

Comment:
This is a table walk, since we don't have an index on `transactions.tx`.  (Or maybe we do?)  I have been thinking of using a different table format, but need to benchmark some things first, so I'm okay with this for now.  (It'll get better in the newer format.)